ledger-tool/src/main.rs

```diff
@@ -2965,3 +2996,110 @@ fn main() {
     measure_total_execution_time.stop();
     info!("{}", measure_total_execution_time);
 }
+
+fn record_transactions(
+    recv: crossbeam_channel::Receiver<TransactionStatusMessage>,
+    filename: &Path,
+) {
+    #[derive(Serialize, Deserialize, Default, Clone)]
+    struct Transaction {
+        index: usize,
+        accounts: Vec<String>,
+        instructions: Vec<Instruction>,
+        is_simple_vote_tx: bool,
+        execution_results: Option<TransactionExecutionDetails>,
+    }
+
+    #[derive(Serialize, Deserialize, Clone)]
+    struct Instruction {
```

Comment:
I'm guessing not, but just making sure, there aren't any suitable pre-existing types to use here are there ?

Reply:
there's some stuff in `solana-transaction-status` that might do the trick

Reply:
All I can find is `solana-transaction-status::parse_instruction::parse` but that only works for a few select program ids, else I get `ParseInstructionError::ProgramNotParsable`, see https://github.com/anza-xyz/agave/blob/master/transaction-status/src/parse_instruction.rs#L118

Reply:
@t-nelson you're right, `UiInstruction` is perfect for this. It gives a very nice result, thanks!